### PR TITLE
Collapse new nodes

### PR DESCRIPTION
**Pull request by martinfleis**
This allows for collapsing of nearby nodes within the newly generated geometry. It ensures that Voronoi-based nodes that look like a single node are collapsed to actual single node.

It is a bit tricky to get the tolerance right and consolidation still works within the whole dbscan cluster rather than iteratively, but it works okay for our use cases.

the consolidation algo tends to fail with larger tolerances on whole networks as the difference call in there occasionally produces MultiLineString but that is for future someone to resolve.

This is all enabled by a change of consolidation algo that can now preserve location of dead-end nodes, which is what we need within the Voronoi to ensure proper re-connection to the rest of the network.

### CODE REVIEW

File: core/algorithms/artifacts.py [hunk 1]
```diff
@@ -1003,6 +1022,10 @@ def nx_gx_cluster(
     else:
         # a loop that has only a single entry point - use individual segments
         merged_edges = edges_on_boundary.dissolve().line_merge().item()
+        if merged_edges.geom_type != "LineString":
+            # this is a fallback for corner cases. It should result in the nearly the
```
Comment (jGaboardi) [inline core/algorithms/artifacts.py:1026]:
```suggestion
            # this is a fallback for corner cases. It should result in nearly the
```

**Review by jGaboardi — APPROVED** [review]
Awesome work! Looks great with a minor typo.

**Comment by anastassiavybornova** [comment]
From the `pipeline` notebook:

```python
new_roads = algorithms.simplify.simplify_singletons(singles, roads)
new_roads = algorithms.simplify.simplify_pairs(doubles, roads)
new_roads = algorithms.simplify.simplify_clusters(clusters, new_roads)
```

2nd line, 2nd argument should be `new_roads`, no?

**Review by anastassiavybornova — APPROVED** [review]
<3

**Comment by martinfleis** [comment]
> 2nd line, 2nd argument should be new_roads, no?

yes, a remainder from debugging process...

**Pull request merged**